public/app/percona/dbaas/components/DBCluster/AddDBClusterModal/AddDBClusterModal.utils.ts

```diff
@@ -0,0 +1,47 @@
+import { getActiveOperators, getDatabaseOptionFromOperator } from '../../Kubernetes/Kubernetes.utils';
+import { AddDBClusterFields, AddDbClusterFormValues } from './AddDBClusterModal.types';
+import { INITIAL_VALUES } from './DBClusterAdvancedOptions/DBClusterAdvancedOptions.constants';
+import { getKubernetesOptions } from './DBClusterBasicOptions/DBClusterBasicOptions.utils';
+import { Kubernetes } from '../../Kubernetes/Kubernetes.types';
+
+export const getInitialValues = (kubernetes: Kubernetes[]): AddDbClusterFormValues => {
+  const activeOperators = getActiveOperators(kubernetes);
+
+  const initialValues: AddDbClusterFormValues = {
+    ...INITIAL_VALUES,
+    [AddDBClusterFields.databaseType]:
+      activeOperators.length === 1
+        ? getDatabaseOptionFromOperator(activeOperators[0])
+        : { value: undefined, label: undefined },
+  };
+
+  if (kubernetes.length > 0) {
+    const kubernetesOptions = getKubernetesOptions(kubernetes);
+    const initialCluster = kubernetesOptions.length > 0 && kubernetesOptions[0];
+    if (initialCluster) {
+      initialValues[AddDBClusterFields.kubernetesCluster] = initialCluster;
+      if (activeOperators.length > 0) {
+        const databaseDefaultOperator = getDatabaseOptionFromOperator(activeOperators[0]);
+        initialValues[AddDBClusterFields.databaseType] = databaseDefaultOperator;
+        initialValues[AddDBClusterFields.name] = `${databaseDefaultOperator?.value}-${generateUID()}`;
+      }
+    }
+  }
+  return initialValues;
+};
+
+export const generateUID = (): string => {
```

Comment:
What happens if there's a conflict with an existing name? (I know that the possibility is low, but will the form still be submittable?)

Reply:
Yeah, good question. I think it would be better to exclude this by checking for matches with existing clusters (something like this - (https://stackoverflow.com/questions/6248666/how-to-generate-short-uid-like-ax4j9z-in-js) first response, second code insertion). I think I should check there is not one of the same name in the list of incoming dbClusters during generation, instead of not letting the user send the form. What do you think, @matejkubinec ?

Reply:
I'm not sure if it's worth the effort of checking for colisions there, if the probability for colision is really small.

Reply:
Thinking about it, the current solution should be good enough.